rewrite-java/src/main/java/org/openrewrite/java/JavaParser.java

```diff
@@ -73,28 +73,32 @@ static List<Path> runtimeClasspath() {
      * matching jars can be found.
      */
     static List<Path> dependenciesFromClasspath(String... artifactNames) {
-        Map<String, Pattern> artifactNamePatterns = Arrays.stream(artifactNames)
-                .collect(toMap(Function.identity(), name -> Pattern.compile(name + "-.*?\\.jar$")));
-
-        Set<String> foundArtifacts = new HashSet<>();
-        List<Path> artifacts = new ArrayList<>();
         List<URI> runtimeClasspath = new ClassGraph().getClasspathURIs();
-        for (URI cpEntry : runtimeClasspath) {
-            for (Map.Entry<String, Pattern> artifactNamePattern : artifactNamePatterns.entrySet()) {
-                if (artifactNamePattern.getValue().matcher(cpEntry.toString()).find()) {
+        List<Path> artifacts = new ArrayList<>(artifactNames.length);
+        List<String> missingArtifactNames = new ArrayList<>(artifactNames.length);
+        for (String artifactName : artifactNames) {
+            Pattern jarPattern = Pattern.compile(artifactName + "-.*?\\.jar$");
+            // In a multiproject IDE classpath, some classpath entries aren't jars
+            Pattern explodedPattern = Pattern.compile("/" + artifactName + "/");
+            boolean lacking = true;
+            for (URI cpEntry : runtimeClasspath) {
+                String cpEntryString = cpEntry.toString();
+                if (jarPattern.matcher(cpEntryString).find()
```

Comment:
I think the logic needs to change slightly, I believe it currently is:

`(isJar or isExploded && isDirectory)`

and we would want this to be : 

`(isJar or (isExploded && isDirectory))`

Reply:
I think once we correct that, we are good. Thank you for your contribution!

Reply:
I just went ahead and fixed this, will wait for tests and then merge.